src/backend/src/pct.jl

```diff
@@ -0,0 +1,175 @@
+    scenariod_variables = unique(last.(split.(prefixed_names, '_')))
+    variables_without_scenario = [
+     "all",
+     "bicycle",
+     "foot",
+     "car_driver",
+     "car_passenger",
+     "motorbike",
+     "train_tube",
+     "bus",
+     "taxi_other",
+     "e_dist_km",
+     "rf_dist_km",
+     "rq_dist_km",
+     "dist_rf_e",
+     "dist_rq_rf",
+     "rf_avslope_perc",
+     "rq_avslope_perc",
+     "rf_time_min",
+     "rq_time_min"]
+
+    df = df[!, ["geo_code1", "geo_code2", prefixed_names..., variables_without_scenario...]]
+
+    vars = vcat(scenariod_variables, variables_without_scenario)
+    zones = sort(unique(vcat(df[:, 1], df[:, 2])))
+    zone_id(code) = findfirst(==(code), zones)
+
+    df[!, :origin] = zone_id.(df[!, 1])
+    df[!, :destination] = zone_id.(df[!, 2])
+
+    df = df[!, ["origin", "destination", prefixed_names..., variables_without_scenario...]]
+
+    column_vars = [ vars[findfirst(v -> endswith(name, v), vars)] for name in names(df)[3:end] ]
+
+    column_scens = map(names(df)[3:end]) do name
+        scen = findfirst(v -> startswith(name, v), scenario_prefixes)
+        isnothing(scen) ? "base" : scenario_prefixes[scen]
+    end
+
+    return ODData(df, column_vars, column_scens)
+end
+
+# Minimum metadata
+function load_pct_metadata(data)
+    # Problems:
+    # Software expects that every scenario contains every column
+    # Solution:
+    # scenarios_with(variable_name)
+    # variables_in(scenario_name)
+    var_defaults = Dict(
+        "good" => "smaller",
+        "thickness" => "variable",
+        "statistics" => "hide",
+        "force_bounds" => [],
+    )
+
+    vars = OrderedDict(var => merge(var_defaults, Dict("scenarios_with" => scenarios_with(data, var))) for var in variables(data))
+
+    scens = OrderedDict(name => Dict("name" => name, "at" => [2010]) for name in scenarios(data))
+
+    OrderedDict(
+        "name" => "PCT stuff",
+        "description" => "blah",
+        "od_matrices" => Dict(
+             "columns" => vars
+        ),
+        "scenarios" => scens,
+    )
+end
+
+import GeoJSON
+import Turf
+
+function load_pct_centroids()
+    zones = GeoJSON.parsefile("$(@__DIR__)/../data/processed/zones.geojson")
+    zone_centroids = Array{Array{Float64,1},1}(undef,length(zones.features))
+    for f in zones.features
+        zone_centroids[f.properties["fid"]] = Turf.centroid(f.geometry).coordinates
+    end
+    return zone_centroids
+end
+
+# Expects to find shapefiles and CSV in `$dir/raw`
+function process_pct_geometry(dir="$(@__DIR__)/../data/")
+    # Get zone names and IDs
+    df = CSV.read(joinpath(dir, "raw/PCT example data commute-msoa-nottinghamshire-od_attributes.csv"), DataFrame; missingstring="NA")
+    zones = sort!(unique(vcat(df[!, :geo_code1], df[!, :geo_code2])))
+    zone_id(code) = findfirst(==(code), zones)
+    all_zones = Dict(zip(zones, keys(zones)))
+
+    shapefile = "$dir/raw/Middle_Layer_Super_Output_Areas__December_2011__EW_BGC_V2.shp"
```

Comment:
@markjd84 if we convert this to geojson, is the result proprietary? Trying to nail down a bug in GeoJSON.jl and wondering if, if I really can't work it out, I can share that file publicly so GeoJSON.jl can fix the bug : )

Reply:
Wait I only just seen this 4 months later... sorry! :-/ If it still applies - I'm happy to do that with odd bits of `pct.jl` or whatever, I'm just not close enough to what you were doing here to fully get which bits are useful more widely :)

Reply:
Haha, no worries. It's not about useful code that we want to share - it's that the data, the shapefile, causes GeoJSON.jl to break in an interesting way when it is read as a `.geojson`. So I want to share that data with GeoJSON.jl in public so that they can see if they can fix their bug.

I can barely remember what the issue was though, so it's not a big priority now :)